modules/common/src/main/scala/notebook/front/gadgets/SparkMonitor.scala

```diff
@@ -19,6 +19,32 @@ class SparkMonitor(sparkContext:SparkContext, checkInterval:Long = 1000) {
 
   sparkContext.listenerBus.addListener(listener)
 
+  /**
+    * Identify link to Spark UI. Supports `yarn-*` modes so far.
+    */
+  def sparkUiLink: Option[String] = sparkContext.master match {
+    case m if m.startsWith("yarn") =>
+      sys.env.get("YARN_JOB_PROXY_URL")
+        .map(yarnProxyURL => s"${yarnProxyURL}/${sparkContext.applicationId}")
+    case _ =>
+      try {
+        val u = sparkContext.getClass.getMethod("ui")
+        u.setAccessible(true)
+        val ui = u.invoke(sparkContext).asInstanceOf[Option[Any]]
+        val appUIAddress:Option[String] = ui.map{ u =>
+          val a = u.getClass.getMethod("appUIAddress")
+          a.setAccessible(true)
+          val appUIAddress = a.invoke(u)
+          appUIAddress.toString
+        }
+        appUIAddress
+      } catch {
+        case e =>
```

Comment:
note this would catch system exceptions like Interrupts, but its ~okish as is, this isn't called that often.